core/src/main/java/org/apache/calcite/rel/externalize/RelJson.java

```diff
@@ -776,4 +905,22 @@ private void addRexFieldCollationList(
     map.put("syntax", operator.getSyntax().toString());
     return map;
   }
+
+  /**
```

Comment:
I wouldn't mention the 'functional interface'. Document its purpose, which is to translate.

Rename `apply` to something like `translate`. `apply` only made sense when this was a `BiFunction`.

In the method javadoc change `Lambda that defines how to transform` to `Transforms`. Less is more.

Rename `stringObjectMap` to `map`.